### PR TITLE
fix: FTL-15464 fetch proper public key for authProvider token validation

### PR DESCRIPTION
**Pull request by rbrazhnyk**

### CODE REVIEW

File: packages/auth-provider/helpers/jwt.ts [hunk 1]
```diff
@@ -31,4 +29,24 @@ export class Jwt {
       return { isValid: false, isExpired: false }
     }
   }
+
+  public async fetchPublicKey(apiGatewayUrl: string): Promise<string> {
+    const { data } = await axios(`${apiGatewayUrl}/iam/.well-known/jwks.json`, {
```
Comment (carlos-affinidi) [inline packages/auth-provider/helpers/jwt.ts:34]: Could we cache/store this key in memory so that we don't have to keep calling this endpoint?

**Review by carlos-affinidi — CHANGES_REQUESTED** [review]

File: packages/auth-provider/helpers/jwt.ts (hunk 1 above)
Comment (rbrazhnyk) [reply packages/auth-provider/helpers/jwt.ts:34]: @carlos-affinidi, it is done ([link to code](https://github.com/affinidi/affinidi-tdk/pull/116/files#diff-8360a309f8b94eeea17081b40c973a080375f593e2dea891741c8d1f9564aba9R82-R84)) in the same way we did for the projectScopedToken - we call fetchPublicKey only if publicKey was never initialised (empty string by default).

Comment (carlos-affinidi) [reply packages/auth-provider/helpers/jwt.ts:34]: Ah, I missed that. Thanks

**Review by carlos-affinidi — APPROVED** [review]

**Pull request merged**